here is the old stuff:
http://sources.redhat.com/ml/libc-alpha/2003-06/msg00027.html
http://www.m17n.org/linux-sh/ml/linux-sh/2003-05/msg00010.html
http://rpm.sh-linux.org/rpm-2004/target/SRPMS/glibc-2.3.3-27.12.src.rpm
http://bugs.gentoo.org/100696

and here is the change that was finally merged:
http://sourceware.org/ml/libc-alpha/2009-01/msg00034.html
http://sourceware.org/ml/libc-alpha/2009-02/msg00000.html

--- a/sysdeps/unix/sysv/linux/sh/Makefile
+++ b/sysdeps/unix/sysv/linux/sh/Makefile
@@ -5,3 +5,8 @@ endif
 ifeq ($(subdir),stdlib)
 gen-as-const-headers += ucontext_i.sym
 endif
+
+ifeq ($(subdir),math)
+# The libm.so link can't find __fpscr_values
+libm.so-no-z-defs = yes
+endif
